File: core/tests/fixtures/analytical/src/py/ledger.py
"""The ledger: core record keeping (fixture)."""


class Ledger:
    """Records entries; the root of the import graph fixture."""

    def __init__(self) -> None:
        self.entries: list[tuple[str, int]] = []

    def record(self, name: str, amount: int) -> None:
        self.entries.append((name, amount))

    def total(self) -> int:
        return sum(amount for _, amount in self.entries)

File: core/tests/fixtures/analytical/src/java/com/acme/core/Ledger.java
package com.acme.core;

import java.util.ArrayList;
import java.util.List;

public class Ledger {
    private final List<Integer> entries = new ArrayList<>();

    public void record(int amount) {
        entries.add(amount);
    }

    public int total() {
        return entries.stream().mapToInt(Integer::intValue).sum();
    }
}
